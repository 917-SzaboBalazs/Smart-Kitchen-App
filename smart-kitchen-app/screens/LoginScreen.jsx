import React, { useState } from 'react';
import { TextInput, Text, StyleSheet, KeyboardAvoidingView, Pressable } from 'react-native';
import colors from '../constants/colors';
import { signInWithEmailAndPassword } from 'firebase/auth';
import { auth } from '../firebase';
import Toast from 'react-native-toast-message';

const LoginScreen = ({ navigation }) => {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');

    const handleLogin = () => {

        signInWithEmailAndPassword(auth, email, password)
            .then(() => {
                navigation.pop();
                navigation.replace("Home");
                resetFields();
            })
            .catch((error) => {
                const errorMessage = error.message;

                console.log(error);


                //alert(errorMessage);

                Toast.show({
                    type: 'error',
                    text1: 'Error',
                    text2: errorMessage,
                });
            });
    };

    const resetFields = () => {
        setEmail('');
        setPassword('');
    }

    return (
        <KeyboardAvoidingView style={styles.container}>
            <Text style={styles.title}>Log In</Text>

            <TextInput
                style={styles.input}
                placeholder="Email"
                value={email}
                onChangeText={setEmail}
                autoCapitalize='none'
            />
            <TextInput
                style={styles.input}
                placeholder="Password"
                secureTextEntry
                value={password}
                onChangeText={setPassword}
                autoCapitalize='none'
            />
            <Pressable
                onPress={handleLogin}
                style={styles.loginButton}
            >
                <Text style={styles.logInText}>Log In</Text>
            </Pressable>
        </KeyboardAvoidingView>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: colors.background,
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        padding: 32,
    },
    title: {
        fontSize: 24,
        fontWeight: 'bold',
        color: colors.primary,
        marginBottom: 24,
        alignSelf: 'flex-start',
    },
    input: {
        width: '100%',
        height: 40,
        borderColor: colors.secondary,
        borderBottomWidth: 1,
        marginBottom: 12,
        paddingHorizontal: 8,
        
    },
    loginButton: {
        width: 100,
        height: 40,
        marginTop: 20,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: colors.secondary,
    },
    logInText: {
        color: 'white',
    },
});

export default LoginScreen;