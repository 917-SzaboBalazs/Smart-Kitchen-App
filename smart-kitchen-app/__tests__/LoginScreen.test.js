import React from 'react';
import renderer from 'react-test-renderer';

import LoginScreen from '../screens/LoginScreen';

describe('<LoginScreen />', () => {
    it('has 4 elements', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree.children.length).toBe(4);
    });
    
    it('renders correctly', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders title correctly', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree.children[0].children[0]).toBe('Log In');
    });

    it('renders email input correctly', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree.children[1].props.placeholder).toBe('Email');
    });

    it('renders password input correctly', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree.children[2].props.placeholder).toBe('Password');
    });

    it('renders login button correctly', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree.children[3].type).toBe('View');
    });
});
