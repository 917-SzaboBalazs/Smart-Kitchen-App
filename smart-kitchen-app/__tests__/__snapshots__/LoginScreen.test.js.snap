// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F7F6BB",
      "flex": 1,
      "justifyContent": "center",
      "padding": 32,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#114232",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 24,
      }
    }
  >
    Log In
  </Text>
  <TextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Email"
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#87A922",
        "height": 40,
        "marginBottom": 12,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value=""
  />
  <TextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#87A922",
        "height": 40,
        "marginBottom": 12,
        "paddingHorizontal": 8,
        "width": "100%",
      }
    }
    value=""
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#87A922",
        "height": 40,
        "justifyContent": "center",
        "marginTop": 20,
        "width": 100,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
        }
      }
    >
      Log In
    </Text>
  </View>
</View>
`;
